modules/post/linux/gather/enum_containers.rb

```diff
@@ -0,0 +1,94 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Linux Container Enumeration',
+      'Description' => %q{
+        This module attempts to enumerate containers running on the target machine.
+        Supports Docker, LXC and RKT.
+      },
+      'License' => MSF_LICENSE,
+      'Author' => ['Mat Rollings [stealthcopter]'],
+      'Platform' => ['linux'],
+      'SessionTypes' => ['shell', 'meterpreter']
+    ))
+  end
+
+  # Check if a container program is installed and usable by current user
+  def runnable(container_type)
+    case container_type
+    when 'docker'
+      result = cmd_exec('docker ps >/dev/null 2>&1 && echo true')
+    when 'lxc'
+      result = cmd_exec('lxc list >/dev/null 2>&1 && echo true')
+    when 'rkt'
+      result = cmd_exec('rkt list >/dev/null 2>&1 && echo true')
+    else
+      print_error("Invalid container type #{container_type}")
+      return false
+    end
+    result =~ /true$/
+  end
+
+  # Count the number of currently running containers
+  def count_containers(container_type)
+    case container_type
+    when 'docker'
+      result = cmd_exec('docker ps --format "{{.Names}}" 2>/dev/null | wc -l')
+    when 'lxc'
+      result = cmd_exec('lxc list -c n --format csv 2>/dev/null | wc -l')
+    when 'rkt'
+      result = cmd_exec('rkt list 2>/dev/null | tail -n +2  | wc -l')
+    else
+      print_error("Invalid container type '#{container_type}'")
+      return 0
+    end
+    result.to_i
+  end
+
+  # List the currently running containers
+  def list_containers(container_type)
+    case container_type
+    when 'docker'
+      result = cmd_exec('docker ps')
```

Comment:
Considering that this is the old format of command for Docker, and that Docker now has the `docker container ls` command, would it make more sense to use `docker container ls` here to prevent risk of the old command being depreciated/removed? Or am I mistaken in saying this?

Reply:
@gwillcox-r7 Is there a reference link for `docker ps` being deprecated? 👀

Reply:
Oh cool, was unaware of this!

Reply:
@adfoster-r7 Not sure if its depreciated right now but Docker is moving to management commands in the format of `docker <management command> <subcommand to run>`. See https://nickjanetakis.com/blog/docker-tip-24-docker-ps-vs-docker-container-ls for more info on this in the context of the `docker ps` command. The `docker ps` command will most likely be depreciated down the road due to the inherent management issues it has vs the management command style.